### PR TITLE
[search] Fixed locality matching code.

### PR DESCRIPTION
**Pull request by ygorshenin**
https://jira.mail.ru/browse/MAPSME-2200

@mpimenov PTAL

### CODE REVIEW

**Comment by ygorshenin** [comment]
Added support for villages localities matching.

File: search/locality_finder.hpp [hunk 1]
```diff
@@ -7,52 +7,59 @@
...
 
   Index const & m_index;
-  MwmSet::MwmId m_worldId;
+  VillagesCache & m_villagesCache;
   unique_ptr<RankTable> m_ranks;
 
-  Cache m_caches[10];
+  m4::Tree<bool> m_coverage;
+  m4::Tree<Item> m_localities;
+  map<MwmSet::MwmId, unordered_set<uint32_t>> m_loadedIds;
```
Comment (ygorshenin) [inline search/locality_finder.hpp]: No, thanks. This is a compressed set of feature ids, not mwm ids.

File: search/locality_finder.cpp [hunk 2]
```diff
@@ -16,20 +18,55 @@ namespace search
...
+  CBV m_cbv;
+};
 
 class DoLoader
 {
 public:
-  DoLoader(MwmContext const & ctx, LocalityFinder::Cache & cache, RankTable const & ranks,
-           int8_t lang)
-    : m_ctx(ctx), m_cache(cache), m_ranks(ranks), m_lang(lang)
+  DoLoader(MwmContext const & ctx, Filter const & filter, int8_t lang,
```
Comment (ygorshenin) [inline search/locality_finder.cpp]: Old name, I left it as is.

File: search/locality_finder.cpp [hunk 3]
```diff
@@ -43,77 +80,53 @@ class DoLoader
     switch (IsLocalityChecker::Instance().GetType(ft))
     {
     case CITY:
-    case TOWN: break;
+    case TOWN:
+    case VILLAGE:
+      break;
     default:  // cache only cities and towns at this moment
```
Comment (ygorshenin) [inline search/locality_finder.cpp]: Done

File: search/search_tests/locality_finder_test.cpp [hunk 4]
```diff
@@ -4,28 +4,38 @@
...
 #include "platform/country_file.hpp"
 #include "platform/local_country_file.hpp"
 #include "platform/local_country_file_utils.hpp"
 #include "platform/platform.hpp"
 
+#include "base/cancellable.hpp"
 
 namespace
 {
+struct TestWithClassifier
```
Comment (ygorshenin) [inline search/search_tests/locality_finder_test.cpp]: Done

File: base/time_samples.cpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+void TimeSamples::Add(double s)
+{
+  m_s += s;
+  m_s2 += s * s;
+  ++m_total;
+}
+
+double TimeSamples::GetMean() const { return m_total == 0 ? 0.0 : m_s / m_total; }
+
+double TimeSamples::GetSd() const
```
Comment (ygorshenin) [inline base/time_samples.cpp]: Why? S means std, d means dev.

File: base/time_samples.hpp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+#include "base/timer.hpp"
+
+#include "std/cstdint.hpp"
+
+namespace my
+{
+class TimeSamples final
+{
+public:
+  void Add(double s);
```
Comment (ygorshenin) [inline base/time_samples.hpp]: seconds. Renamed.

**Comment by ygorshenin** [comment]
@mpimenov PTAL

File: search/locality_finder.hpp (hunk 1 above)
Comment (mpimenov) [inline search/locality_finder.hpp]: loadedFeatureIds?

File: base/time_samples.cpp (hunk 5 above)
Comment (mpimenov) [inline base/time_samples.cpp]: It's just more common.

Comment (ygorshenin) [inline base/time_samples.cpp]: https://stat.ethz.ch/R-manual/R-devel/library/stats/html/sd.html

Where it's more common and recognizable? The R community disagree with you.

**Comment by mpimenov** [comment]
LGTM

**Pull request merged**